src/SonarQube.Client.Tests/Api/SonarQubeService_GetAllModulesAsync.cs

```diff
@@ -0,0 +1,155 @@
+
+            SetupRequest("api/components/tree?qualifiers=BRC&component=myProject&p=1&ps=500", @"
+{
+  ""paging"": {
+    ""pageIndex"": 1,
+    ""pageSize"": 100,
+    ""total"": 2
+  },
+  ""baseComponent"": {
+    ""key"": ""sq-project-key"",
+    ""name"": ""SonarQube Project Name"",
+    ""qualifier"": ""TRK""
+  },
+  ""components"": [
+    {
+      ""key"": ""sq-project-key:sq-project-key:50934C7A-2751-4675-91C4-CFD37C8BF57C"",
+      ""name"": ""Project1"",
+      ""qualifier"": ""BRC"",
+      ""path"": ""src/Project1""
+    },
+    {
+      ""key"": ""sq-project-key:sq-project-key:39CCD086-A7F8-42A0-B402-3C9BD9EB4825"",
+      ""name"": ""Project2"",
+      ""qualifier"": ""BRC"",
+      ""path"": ""src/Project2""
+    }
+  ]
+}");
+
+            var result = await service.GetAllModulesAsync("myProject", CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(3);
+
+            result[0].Key.Should().Be("sq-project-key");
+            result[0].Name.Should().Be("SonarQube Project Name");
+            result[0].RelativePathToRoot.Should().BeNull();
+
+            result[1].Key.Should().Be("sq-project-key:sq-project-key:50934C7A-2751-4675-91C4-CFD37C8BF57C");
+            result[1].Name.Should().Be("Project1");
+            result[1].RelativePathToRoot.Should().Be("src/Project1");
+
+            result[2].Key.Should().Be("sq-project-key:sq-project-key:39CCD086-A7F8-42A0-B402-3C9BD9EB4825");
+            result[2].Name.Should().Be("Project2");
+            result[2].RelativePathToRoot.Should().Be("src/Project2");
+        }
+
+        [TestMethod]
+        public async Task GetModules_SonarQubeProjectWithOneCSharpProject()
+        {
+            await ConnectToSonarQube();
+
+            SetupRequest("api/components/tree?qualifiers=BRC&component=myProject&p=1&ps=500", @"
+{
+  ""paging"": {
+    ""pageIndex"": 1,
+    ""pageSize"": 100,
+    ""total"": 1
+  },
+  ""baseComponent"": {
+    ""key"": ""sq-project-key"",
+    ""name"": ""SonarQube Project Name"",
+    ""qualifier"": ""TRK""
+  },
+  ""components"": [
+    {
+      ""key"": ""sq-project-key:sq-project-key:50934C7A-2751-4675-91C4-CFD37C8BF57C"",
+      ""name"": ""Project1"",
+      ""qualifier"": ""BRC""
+    }
+  ]
+}");
+
+            var result = await service.GetAllModulesAsync("myProject", CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(2);
+
+            result[0].Key.Should().Be("sq-project-key");
+            result[0].Name.Should().Be("SonarQube Project Name");
+            result[0].RelativePathToRoot.Should().BeNull();
+
+            result[1].Key.Should().Be("sq-project-key:sq-project-key:50934C7A-2751-4675-91C4-CFD37C8BF57C");
+            result[1].Name.Should().Be("Project1");
+            result[1].RelativePathToRoot.Should().BeNull();            
+        }
+
+        [TestMethod]
+        public async Task GetModules_ExpectedNewSonarQubeApi()
```

Comment:
This test name is a bit confusing. What is the test checking that the previous two don't - that the API handles a response that only has a root project?

Reply:
According to Duarte, when a project will no longer have modules (only the root) we expect the API to return the following json (`baseComponent` filled and `components` empty).